From 12f883d3601bb695e589ab8b570d81d789c4bf64 Mon Sep 17 00:00:00 2001
From: Eike Rathke <erack@redhat.com>
Date: Sun, 22 Jan 2012 20:13:36 +0100
Subject: [PATCH 3/5] resolved fdo#45032 Calc export to HTML with graphics
 failed

Fixed a misconception in transition from List to to ptr_vector of commit
db599f64e63a3141159b7f0e59783197c39b3e02, List::Next() returned NULL when at
end of list, mimic that.

Signed-off-by: Kohei Yoshida <kohei.yoshida@suse.com>
---
 sc/source/filter/html/htmlexp.cxx |    9 ++++++---
 1 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/sc/source/filter/html/htmlexp.cxx b/sc/source/filter/html/htmlexp.cxx
index 772b72b..34e994b 100644
--- a/sc/source/filter/html/htmlexp.cxx
+++ b/sc/source/filter/html/htmlexp.cxx
@@ -829,11 +829,14 @@ void ScHTMLExport::WriteCell( SCCOL nCol, SCROW nRow, SCTAB nTab )
         size_t ListSize = aGraphList.size();
         for ( size_t i = 0; i < ListSize; ++i )
         {
-            pGraphEntry = &aGraphList[ i ];
-            if ( pGraphEntry->bInCell && pGraphEntry->aRange.In( aPos ) )
+            ScHTMLGraphEntry* pE = &aGraphList[ i ];
+            if ( pE->bInCell && pE->aRange.In( aPos ) )
             {
-                if ( pGraphEntry->aRange.aStart == aPos )
+                if ( pE->aRange.aStart == aPos )
+                {
+                    pGraphEntry = pE;
                     break;  // for
+                }
                 else
                     return ;		// ist ein Col/RowSpan, Overlapped
             }
-- 
1.7.7.6

